src/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring/Windows/Interop/ProcessInfo.cs

```diff
@@ -1,6 +1,7 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
```

Comment:
Please update WindowsSnapshotProvider class with the same change as well

Reply:
@evgenyfedorov2 @danespinosa WindowsSnapshotProvider is creating snapshot for (`PrivilegedProcessorTime`, `UserProcessorTime`) which requires `Process`:
```
    public Snapshot GetSnapshot()
    {
        using var process = Process.GetCurrentProcess();

        return new Snapshot(
            totalTimeSinceStart: TimeSpan.FromTicks(_timeProvider.GetUtcNow().Ticks),
            kernelTimeSinceStart: process.PrivilegedProcessorTime,
            userTimeSinceStart: process.UserProcessorTime,
            memoryUsageInBytes: (ulong)process.WorkingSet64);
    }
```
The replacement is `Environment.CpuUsage.PrivilegedTime` and `Environment.CpuUsage.UserTime` from [Environment.CpuUsage](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Environment.Windows.cs#L373), `Process` will leverage this as well: [](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Windows.cs#L235).
![image](https://github.com/user-attachments/assets/056266d2-d7c1-4253-ad3a-690d0dfa0e35)
However, these are only been supported in .NET 9:
![image](https://github.com/user-attachments/assets/29572db9-fb85-4528-b5c1-894a6fd57007)
![image](https://github.com/user-attachments/assets/6c325d15-b5ab-4ffa-83fd-8f2db4e36bb2)

Reply:
Thanks, that would be too much for this PR. Please create an issue for it instead.

But actually what I was asking about is this https://github.com/dotnet/extensions/blob/3f1f59cd52c58ab7e5666ba4c8ea6176f606f3d7/src/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring/Windows/WindowsSnapshotProvider.cs#L110 - can be replaced with Environment.WorkingSet, right?

Reply:
Sure, updated.

Reply:
Thank you!